nixjs-rt/src/lib.ts

```diff
@@ -227,15 +245,33 @@ export abstract class NixType {
    */
   abstract typeOf(): string;
 
+  /**
+   * Returns a human-readable string representation of this value, that can be inserted into a sentence.
+   *
+   * For example, "a string", "an array", etc.
+   *
+   * Static functions can't be made abstract, so abstract is omitted here.
+   */
+  static toHumanReadable(): string {
+    throw new Error("abstract");
+  }
+
+  /**
+   * Returns the name of the type (as a string, which effectively acts as an enum).
+   *
+   * This is used for identifying types for error messages.
+   */
+  static toTypeName(): NixTypeName {
```

Comment:
How come you're creating a new function instead not using the `typeOf` function?

Reply:
That one's static, I'm not sure how to make a function work on both static and instance using a single function definition